base/math.jl

```diff
@@ -851,26 +851,58 @@ minmax(x::T, y::T) where {T<:AbstractFloat} = min(x, y), max(x, y)
 
 _isless(x::Float16, y::Float16) = signbit(widen(x) - widen(y))
 
+const _minmaxtypes = [(:f64, :Float64, :double), (:f32, :Float32, :float)] #TODO: Add a way to detect native f16
+const _minmaxops = [(:min, :minimum), (:max, :maximum)]
+for op in _minmaxops
+    jlop,llvmop = op
+    for type in _minmaxtypes
+        fntyp, jltyp, llvmtyp = type
+        @eval @inline function $(Symbol("llvm_", jlop))(x::$jltyp,y::$jltyp)
```

Comment:
does this need effects?

Reply:
I haven't checked, but it's very likely.

Reply:
Is this total? Or foldable + nothrow?

Reply:
should be total. (although I think those are the same)